src/context/CartFunctions.jsx

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable react/prop-types */
+import { useMemo, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { CartContext } from "./CartContext";
+
+export const CartFunctions = ({ children }) => {
```

Comment:
```suggestion
export const CartProvider = ({ children }) => {
```
ending a React Component's name with words like Functions is generally a bad idea